src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs

```diff
@@ -156,10 +157,11 @@ private void AddNewOperation(ulong key, Operation operation)
 
             private unsafe bool WriteSubmissions()
             {
-                Ring ring = _ring!;
+                ConcurrentRing ring = _ring!;
                 int iovIndex = _iovsUsed;
                 int sqesAvailable = ring.SubmissionEntriesAvailable;
```

Comment:
`ring.SubmissionEntriesAvailable` returns a snapshot that could change (increase) while iterating below. We need to find an elegant way to continue submitting, while there are both new operations and enough (1 or 2 depending on Op) SQEs available.